server/src/main/java/io/druid/server/coordinator/DiskNormalizedCostBalancerStrategy.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.coordinator;
+
+import com.google.common.base.Predicates;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.metamx.emitter.EmittingLogger;
+import io.druid.java.util.common.Pair;
+import io.druid.timeline.DataSegment;
+import org.apache.commons.math3.util.FastMath;
+import org.joda.time.Interval;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Executors;
+
+public class DiskNormalizedCostBalancerStrategy extends CostBalancerStrategy
+{
+  public DiskNormalizedCostBalancerStrategy(ListeningExecutorService exec)
+  {
+    super(exec);
+  }
+
+  @Override
+  protected double computeCost(
+      final DataSegment proposalSegment, final ServerHolder server, final boolean includeCurrentServer
+  )
+  {
+    double cost = super.computeCost(proposalSegment, server, includeCurrentServer);
+
+    if(cost == Double.POSITIVE_INFINITY){
+      return cost;
+    }
+
+    int nSegments = 1;
+    if(server.getServer().getSegments().size() > 0)
+    {
+      nSegments = server.getServer().getSegments().size();
+    }
+
+    double normalizedCost = cost/nSegments;
+    double usageRatio = (double)server.getServer().getCurrSize()/(double)server.getServer().getMaxSize();
+
+    return normalizedCost*usageRatio;
```

Comment:
any reason why this is better than the previous one? I guess I'm not sure I fully understand normalization, a comment might be worthwhile.